src/Agent/NewRelic/Agent/Core/DistributedTracing/Samplers/SamplerService.cs

```diff
@@ -0,0 +1,101 @@
+// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using NewRelic.Agent.Configuration;
+using NewRelic.Agent.Core.Events;
+using NewRelic.Agent.Core.Utilities;
+using NewRelic.Agent.Extensions.Logging;
+
+namespace NewRelic.Agent.Core.DistributedTracing.Samplers;
+
+public enum SamplerType
+{
+    Root,
+    RemoteParentSampled,
+    RemoteParentNotSampled,
+}
+
+public interface ISamplerService
+{
+    /// <summary>
+    /// Returns the appropriately configured sampler based on the SamplerType
+    /// </summary>
+    /// <param name="samplerType"></param>
+    /// <returns></returns>
+    /// <exception cref="ArgumentOutOfRangeException"></exception>
+    ISampler GetSampler(SamplerType samplerType);
+}
+
+public class SamplerService : ConfigurationBasedService, ISamplerService
+{
+    private ISampler _rootSampler;
```

Comment:
Rather than making these distinct variables, maybe this should be a Dictionary with the `SamplerType` as the key? That might make the code a little cleaner, and would make it easy to add any new sampler conditions later (Though admittedly there probably won't be any others).